Clean up README.md

MeGaGiGaGon opened the pull request:
I found the `README` very confusing, especially the commands portion, which I had to read the source code to understand. This cleans it up to a more "standard" format, using `<argument>` for required positional arguments.

I removed having the roles inside a quote block `>` since the greying out of the text makes it hard to read on some (my) screens, moving to a list based format `-` instead. ~~Since markdown lists are kind of dumb when it comes to combining, I added the new `####` section headers to separate them.~~

I also fixed some spelling/grammar issues, and re-wrote a few lines to make it more clear, at least to me.

CTNOriginals committed:
Merge pull request #1 from CTN-Originals/development

1.0.1

CTNOriginals committed:
Merge pull request #2 from CTN-Originals/development

Legal documents + prep for public

CTNOriginals committed:
Merge pull request #3 from CTN-Originals/development

1.0.2

MeGaGiGaGon committed:
Clean up README.md

MeGaGiGaGon committed:
Change a few words

CTNOriginals commented:
Thanks for the interest in the project and taking the time to contribute!
I can not accept this yet and would also like to discuss some alternatives before I continue.

As for the layout of the commands initially, the extra lines that were the sub command groups and the subcommands themselves were not excluded as they also have their own description and should not be excluded from the end result as its valiable information for developers and user. Though I do agree with you that it might be a bit confusing as they also started with a `/` and that that might have signified that those were also usable commands. So i would like to hear your take on how to apply this otherwise while not excluding information.

As for the embed format `>` blocks, yeah it is pretty annoying how github displays these blocks and it might indeed be best to leave those out.

On another note, you also added a `####` header in formatting, but this I can not allow as i would like to keep the readme compatible with discords markdown parsing, and discord goes up to 3 sizes only.

Please let me know your thoughts and feel free to alter the readme once again with this knowledge

MeGaGiGaGon committed:
Remove `####`s

MeGaGiGaGon commented:
The reason I removed them is I don't see what information they give, since a command/sub-command on it's own is not a valid command. Ie, `/config admin-role` isn't a valid command. Removing them also gets rid of a lot of visual clutter, since if I'm looking in a `README.md`, I want the commands given to "just work".

This is one way command line tool docs are usually handled, the other being a "tree" of help info. Ie
```
$ RoleMentionCooldown
usage: RoleMentionCooldown <command>

commands:
    config           Configure the bot's settings
    role_cooldown    add/remove/change role cooldowns

$ RoleMentionCooldown config
usage: RoleMentionCooldown config <sub-command>

sub-commands:
    display                            Display the current server configurations
    admin-role <add | remove><role>    Add or remove an admin role
```
This works nicely for the command line, since you can traverse down the help branch you're interested in, but flattening it out to text like the existing `README.md` does leads to unclear/duplicated information.

---

I replaced the `####` with more list nesting

CTNOriginals commented:
Thanks for the clarifications! This now indeed does make more sense and am willing to merge this.
Thanks again for the contribution!
